### PR TITLE
SUOSERIAL Demonstration Example

### PR DESCRIPTION
**Pull request by ioannis-karachalios**
This application should be [ported](https://github.com/dialog-semiconductor/BLE_SDK10_examples/pull/52) from the DA1469x family of devices. Naming convention is remained, despite the fact that the DA1459x family of devices does not feature a USB device controller.

### CODE REVIEW

**Comment by ioannis-karachalios** [comment]
@koen-vdm can you please elaborate on the CI report? It's not clear to me why the CI fails to find the project.

**Comment by koen-vdm** [comment]
Hello @ioannis-karachalios,
Please have a look at [this comment](https://github.com/dialog-semiconductor/BLE_SDK10_DA1459x_examples/pull/4#issuecomment-2202030512). The issue here is similar, only here, the build config is named *-Debug_QSPI.

**Comment by ioannis-karachalios** [comment]
@koen-vdm, thanks for the prompt reply. CI issues are now solved.

**Review by Aart-Kleinendorst — CHANGES_REQUESTED** [review]
Hi Ioannis,
1/ Nice software example
2/ I see no messages in the RTT viewer
3/ Please add instructions how to add the SUOSERIAL (instead of SUOTA) to an existing project 
4/ Please add the SUSERIAL adapted PXP-reporter project (zipped)

**Comment by Aart-Kleinendorst** [comment]
Hi Ioannis,
If possible please try to reduce the (Release) version <= 64 kB so it will also can be used in the asymmetric Software Update.
Br, Aart

**Comment by ioannis-karachalios** [comment]
> Hi Ioannis, If possible please try to reduce the (Release) version <= 64 kB so it will also can be used in the asymmetric Software Update. Br, Aart

I did a quick test and code size can be reduced from  ~69kB to ~65kB if the Segger libraries and WDG service are excluded from build.  Another mechanism that significantly reduces code size is the RCx calibration, by zeroing `dg_configUSE_SYS_ADC`. However, this can be an issue if the device is exposed to high/low temperatures. The latter mechanism occupies over 5kB of code size. The rest of the configurations macros should be defaulted to optimized values. As such, further investigation is required in order to shrink code size by another 1kB or more so we can achieve the required <= 64kB. 
I'd prefer that this topic is discussed separately (in a next code revisit) as asymmetric images have yet to be supported, officially.

**Comment by ioannis-karachalios** [comment]
> 2/ I see no messages in the RTT viewer

Please make sure the RTT session is up and running when SUOSERIAL is performed. Queued messages should be displayed even if the session is enabled after SUOSERIAL session. Here are the logs that should be displayed on the RTT Viewer terminal:

rtt_viewer_suoserial_debug_info

**Comment by ioannis-karachalios** [comment]
> 4/ Please add the SUSERIAL adapted PXP-reporter project (zipped)

The modified proximity reporter source code is pushed under the  `pxp_reporter_suoserial.zip` folder. I removed the `Debug` build rules as they increase code size significantly which results in raising linker errors due to code overflow (the asymmetric image concept is not tested and so it's not considered).

**Comment by ioannis-karachalios** [comment]
> 3/ Please add instructions how to add the SUOSERIAL (instead of SUOTA) to an existing project

README now includes the `Adding Support for the SUOSERIAL Service` section that illustrates all the steps required to build an application with the SUOSERIAL service. In fact, SUOSERIAL and SUOTA can co-exit and be used interchangeably. There is no protection mechanism, though, so it's developer's responsibility to prevent the scenario that both mechanisms are not used at the same time.  In the feature, we could revisit that part and add some primitive protection mechanism.

**Comment by ioannis-karachalios** [comment]
@Aart-Kleinendorst is there any further feedback on this PR? Can we merge it into main?

File: features/suoserial_sample_code/Readme.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+- This example employs the `SUOTA` flash partition so that there is room for both the current and update firmwares. As such, it's imperative that the flash memory be erased before flashing the target application. This is imperative every time a new flash partition is loaded.
+
+- Compile the source code (either in Release or Debug mode) and flash it into the chip. Please note that the debug flavor should be used merely for debugging purposes since it should increase the generated binary file, significantly. In addition, the source code is built to work with the embedded flash Working with external flash memory devices is out of the scope of this demonstration example.  
+
+- Once the application image is flashed, press the RESET button on the daughter board to start executing the application. 
+
+- Open a command prompt at `<path_on_your_machine>/suoserial_sample_code/suoserial_host`. 
+
+- If the host runs on a Windows machine hit `host_usb.exe <COM port> pxp_reporter.1.0.0.1.img -verbose`.
```
Comment (Aart-Kleinendorst) [inline features/suoserial_sample_code/Readme.md:34]: change 'host_usb.exe' in 'host_suoserial.exe'

**Review by Aart-Kleinendorst — CHANGES_REQUESTED** [review]
suoserial_host>host_suoserial.exe COM4 pxp_reporter.1.0.0.1.img FAILS  
HOST_USB_UPDATER_VERSION = 2
Error in opening serial port

Result:Fail

If I disable config_RTT (to reduce code-size a HARDFAULT is generated.

File: features/suoserial_sample_code/Readme.md (hunk 1 above)
Comment (ioannis-karachalios) [reply features/suoserial_sample_code/Readme.md:34]: Done

**Comment by ioannis-karachalios** [comment]
> esult:Fail

Only the serial port number should be provided, that is `4` and not `COM4`. As for the HARDFAULT issue, I am not able to reproduce it. Please note that the RTT is disabled by default, as debug messages are exchanged through the executable application. I have moved the PR onto our internal Bitbucket to follow the official procedure. Closing the current PR.

**Comment by ioannis-karachalios** [comment]
PR has been moved to our internal Bitbucket. Discussions will be continued from there.

**Pull request closed without merging**